warpforge-api/src/formula.rs

```diff
@@ -0,0 +1,41 @@
+use std::ops::Not;
+
+use derive_more::{Display, FromStr};
+use serde::{Deserialize, Serialize};
+
+// FUTURE: Could be represneted as an enum, discriminating on the first char being '/' or '$'
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize, FromStr, Display)]
+pub struct SandboxPort(pub String);
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct GatherDirective {
+	pub from: SandboxPort,
+	pub packtype: Option<crate::content::Packtype>,
+	// TODO:
+	// filters: Option<FilterMap>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum Action {
```

Comment:
I think you can blast away the `compute.rs` file now, as you've now here finished what it was supposed to be.  ^^

Reply:
Blasted away (https://github.com/janispeyer/rustwarp/commit/4049a160e60357ceda6ab319d828343c91ad3d82)